keps/sig-node/20190825-exec-as-specified-user.md

```diff
@@ -0,0 +1,129 @@
+---
+title: Exec as specified user
+authors:
+  - "@max8899"
+owning-sig: sig-node
+participating-sigs:
+  - sig-cli
+  - sig-auth
+reviewers:
+  - "@smarterclayton"
+  - "@dchen1107"
+approvers:
+  - TBD
+editor: TBD
+creation-date: 2019-08-25
+last-updated: 2019-08-25
+status: implemented
+see-also:
+  - "https://github.com/kubernetes/kubernetes/issues/30656"
+  - "https://github.com/kubernetes/kubernetes/pull/81883"
+replaces:
+  - "n/a"
+superseded-by:
+  - "n/a"
+---
+
+# Exec as specified user
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+- [Design Details](#design-details)
+  - [API Specification](#api-specification)
+  - [Library Specification](#library-specification)
+- [Implementation History](#implementation-history)
+<!-- /toc -->
+
+## Summary
+
+As a Kubernetes User, we should be able to specify username for the containers when doing exec, similar to how docker allows that using docker exec options `-u`, 
+```
+-u, --user="" Username or UID (format: <name|uid>[:<group|gid>]) format
+```
+
+## Motivation
+
+The root problem is lacking of standard description for the `exec` operation.
+
+OCI didn't introduce `exec` as a regular operation for container runtime[2],
+
+so when Kubernetes announce [Container Runtime Interface(CRI)](https://kubernetes.io/blog/2016/12/container-runtime-interface-cri-in-kubernetes/) in release 1.5, it became the de-facto standard of `exec`, without a `user` option.
+
+
+>  In the Kubernetes 1.5 release, we are proud to introduce the Container Runtime Interface (CRI) – a plugin interface which enables kubelet to use a wide variety of container runtimes, without the need to recompile. CRI consists of a protocol buffers and gRPC API, and libraries, with additional specifications and tools under active development. CRI is being released as Alpha in Kubernetes 1.5.
+
+But There are many container runtime implementation, e.g Docker, Kata, gVisior, most of which provide a way to enter the container as a special `user`.
+
+It is because the `exec` is basically to start a process inside the container runtime, and the [`process`](https://github.com/opencontainers/runtime-spec/blob/master/specs-go/config.go#L39) has a user definition, which is required.
```

Comment:
I'd suggest to use permanent link instead:
```suggestion
It is because the `exec` is basically to start a process inside the container runtime, and the [`process`](https://github.com/opencontainers/runtime-spec/blob/52e2591aa9f7211d64c49c4fed8691a183189284/specs-go/config.go#L39) has a user definition, which is required.
```